pkg/helm/helmchart_structure_exists.go

```diff
@@ -51,61 +56,52 @@ func (r *HelmStructureExists) ExternalDescription() string {
 	return "https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-02"
 }
 
+func (r *HelmStructureExists) IsOptional() bool {
+	return false
+}
+
 func (r *HelmStructureExists) Test() *tractusx.QualityResult {
-	helmStructureFiles := []string{
-		".helmignore",
-		"Chart.yaml",
-		"LICENSE",
-		"README.md",
-		"values.yaml",
-	}
+	var missingFiles []string
+	var chartYamlFiles []string
+	errorDescriptionCharts := ""
+	chartsValid := true
 
 	chartDir := path.Join(r.baseDir, "charts")
-	if fi, err := os.Stat(chartDir); err != nil || !fi.IsDir() {
-		return &tractusx.QualityResult{Passed: true}
-	}
-
 	helmCharts, err := os.ReadDir(chartDir)
 	if err != nil || len(helmCharts) == 0 {
-		return &tractusx.QualityResult{ErrorDescription: fmt.Sprintf("Can't read Helm Charts at %s.", chartDir)}
+		return &tractusx.QualityResult{Passed: true}
 	}
 
-	var missingFiles []string
-	var chartYamlFiles []string
 	for _, hc := range helmCharts {
-		if hc.IsDir() {
-			for _, fname := range helmStructureFiles {
-				fpath := filepath.Join(chartDir, hc.Name(), fname)
-				isMissing := filesystem.CheckMissingFiles([]string{fpath})
-				if fname == "Chart.yaml" && isMissing == nil {
-					chartYamlFiles = append(chartYamlFiles, fpath)
-				} else if isMissing != nil {
-					missingFiles = append(missingFiles, isMissing...)
-				}
-			}
+		chartYamlPath := path.Join(chartDir, hc.Name(), "Chart.yaml")
+		if _, err := os.Stat(chartYamlPath); err != nil || !hc.IsDir() {
```

Comment:
`continue` in a loop is often very subtle and therefore overseen. 
I think we should either 
a) filter the directories before doing the actual check.. or
b) extract the condition to a function like `isChartDirectory` or `doesContainChart´, so it is a bit easier to understand without closely looking at the code

Reply:
@SebastianBezold thanks for suggestion, extracted the condition to function isChartDirectory